types/bootstrap/js/dist/tooltip.d.ts

```diff
@@ -82,6 +82,11 @@ declare class Tooltip extends BaseComponent {
      * Updates the position of an element’s tooltip.
      */
     update(): void;
+
+    /**
+     * Gives a way to change the tooltip’s content after its initialization.
+     */
+    setContent(content?: any): void;
```

Comment:
> The setContent method accepts an object argument, where each property-key is a valid string selector within the popover template, and each related property-value can be string | element | function | null

https://getbootstrap.com/docs/5.2/components/tooltips/#methods

I think the func is like `() => string | element | null`, though it looks like deceased feature in source code.

Reply:
I think the content is something like this 
```typescript
type SetContentFunction = () => string | element | function | null
Record<string, string | element | SetContentFunction | null>
```

Reply:
I think the content is something like this 
```typescript
type SetContentFunction = () => string | element | function | null
setContent(content?: Record<string, string | element | SetContentFunction | null>): void;
```

Reply:
Hi, I apologize, I don't 100% understand what you are saying but I will do my best.

I made a commit but I'm unsure of the following:
`tooltip.d.ts`
Line 41, Is this class static parameter required or should I just omit it?
Line 328, I'm using `Function` which is banned (error below)

```
.../DefinitelyTyped/types/bootstrap/js/dist/tooltip.d.ts:1:25
ERROR: 1:25    expect     TypeScript@4.8 compile error:
Cannot find module '@popperjs/core' or its corresponding type declarations.
ERROR: 326:56  ban-types  Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
```

Reply:
> you mentioned it being a deceased feature?

I don't recall usage of the function that return a function. I've personally used string markup only

Reply:
I digged further, it seems still supported. I don't know how to reference the code like you did when you started this conversation (please let me know how you did it) so I'm copy pasting it below with references

```js
static get DefaultType() {
    return DefaultType
}
```
https://github.com/twbs/bootstrap/blob/edf9c40956d19e6ab3f9151bfe0dfac6be06fa21/js/src/tooltip.js#L131

```js
const DefaultType = {
  ...Tooltip.DefaultType,
  content: '(null|string|element|function)'
}
```
https://github.com/twbs/bootstrap/blob/edf9c40956d19e6ab3f9151bfe0dfac6be06fa21/js/src/popover.js#L34

```js
  _getTemplateFactory(content) {
    if (this._templateFactory) {
      this._templateFactory.changeContent(content)
    } else {
      this._templateFactory = new TemplateFactory({
        ...this._config,
        // the `content` var has to be after `this._config`
        // to override config.content in case of popover
        content,
        extraClass: this._resolvePossibleFunction(this._config.customClass)
      })
    }

    return this._templateFactory
  }
```
https://github.com/twbs/bootstrap/blob/edf9c40956d19e6ab3f9151bfe0dfac6be06fa21/js/src/tooltip.js#L352

```js
  _getContent() {
    return this._resolvePossibleFunction(this._config.content)
  }
```
https://github.com/twbs/bootstrap/blob/edf9c40956d19e6ab3f9151bfe0dfac6be06fa21/js/src/popover.js#L70

```js
  _resolvePossibleFunction(arg) {
    return typeof arg === 'function' ? arg.call(this._element) : arg
  }
  ```
https://github.com/twbs/bootstrap/blob/edf9c40956d19e6ab3f9151bfe0dfac6be06fa21/js/src/tooltip.js#L413

Reply:
I've only ment: I never saw example of using function return type in BS docs or BS own tests. I know implementaton exists. That's just side topic, can be update in future,
thanks!